extension/chrome/elements/compose-modules/formatters/encrypted-mail-msg-formatter.ts

```diff
@@ -76,21 +78,25 @@ export class EncryptedMsgMailFormatter extends BaseMailFormatter {
       const mimeEncodedPlainMessage = await Mime.encode(msgBody, { Subject: newMsg.subject }, atts);
       const encryptedMessage = await SmimeKey.encryptMessage({ pubkeys: x509certs, data: Buf.fromUtfStr(mimeEncodedPlainMessage) });
       const body = { "encrypted/buf": Buf.fromUint8(encryptedMessage.data) };
-      return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body, type: 'smimeEncrypted', atts: [], isDraft: this.isDraft });
+      //return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body, type: 'smimeEncrypted', atts: [], isDraft: this.isDraft });   
+      return await SendableMsg.createSMime(this.acctEmail, this.headers(newMsg), body, { isDraft: this.isDraft });
     }
     // openpgp
     const atts: Att[] = this.isDraft ? [] : await this.view.attsModule.attach.collectEncryptAtts(pubs);
     const { data: encryptedBody, type } = await this.encryptDataArmor(Buf.fromUtfStr(newMsg.plaintext), undefined, pubs, signingPrv);
     const mimeType = type === 'smime' ? 'smimeEncrypted' : undefined;
-    return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body: { "encrypted/buf": Buf.fromUint8(encryptedBody) }, type: mimeType, atts, isDraft: this.isDraft });
+    //return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body: { "encrypted/buf": Buf.fromUint8(encryptedBody) }, type: mimeType, atts, isDraft: this.isDraft });
+    //return await SendableMsg.createSMimeOrOpenPGP(this.acctEmail, { ...this.headers(newMsg), body: { "encrypted/buf": Buf.fromUint8(encryptedBody) }, type: mimeType, atts, isDraft: this.isDraft });
+    return await SendableMsg.createPgpInline(this.acctEmail, this.headers(newMsg), { "encrypted/buf": Buf.fromUint8(encryptedBody) }, atts, { type: mimeType, isDraft: this.isDraft });
   }
 
   private sendableRichTextMsg = async (newMsg: NewMsgData, pubs: PubkeyResult[], signingPrv?: Key) => {
     const plainAtts = this.isDraft ? [] : await this.view.attsModule.attach.collectAtts();
     const pgpMimeToEncrypt = await Mime.encode({ 'text/plain': newMsg.plaintext, 'text/html': newMsg.plainhtml }, { Subject: newMsg.subject }, plainAtts);
     const { data: encrypted } = await this.encryptDataArmor(Buf.fromUtfStr(pgpMimeToEncrypt), undefined, pubs, signingPrv);
     const atts = this.createPgpMimeAtts(encrypted);
-    return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body: {}, atts, type: 'pgpMimeEncrypted', isDraft: this.isDraft });
+    //return await SendableMsg.create(this.acctEmail, { ...this.headers(newMsg), body: {}, atts, type: 'pgpMimeEncrypted', isDraft: this.isDraft });
+    return await SendableMsg.createPgpMime(this.acctEmail, this.headers(newMsg), {}, atts, { isDraft: this.isDraft });
```

Comment:
Here the body is always empty, and therefore can be removed from the signature. Pgp mime messages always have empty body and content is in attachments.

Reply:
Ok, but can you take a look then at PlainMsgMailFormatter? This looks not correct use of method if body is always empty.

```
...
body['text/html'] = newMsg.plainhtml;
...
return await SendableMsg.createPgpMime(this.acctEmail, this.headers(newMsg), body, atts);
```

Reply:
Ok. The reason why i kept the empty body was because of the next case below that needs to be converted to createPlain. With this new method in place, i removed the empty body.

Reply:
Perfect